testrunner-running/src/main/scala/edu/illinois/cs/xstream/CustomElementIgnoringMapper.java

```diff
@@ -0,0 +1,81 @@
+package edu.illinois.cs.xstream;
+
+import com.thoughtworks.xstream.core.util.FastField;
+import com.thoughtworks.xstream.mapper.ElementIgnoringMapper;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+public class CustomElementIgnoringMapper extends ElementIgnoringMapper {
+    public CustomElementIgnoringMapper(Mapper mapper) {
+        super(mapper);
+    }
+
+    @Override
+    public boolean shouldSerializeMember(final Class definedIn, final String fieldName) {
+        if (fieldsToOmit.contains(customKey(definedIn, fieldName))) {
+            return false;
+        } else if (definedIn == Object.class && isIgnoredElement(fieldName)) {
+            return false;
+        }
+        try {
+            // Hack to ignore field of type CodeSource
+            System.out.println("PROCESS: " + definedIn.getName() + " - " + definedIn.getDeclaredField(fieldName).getType() + " - " + fieldName);
+
+            if (definedIn.getDeclaredField(fieldName).getType().equals(Class.forName("java.security.CodeSource"))) {
```

Comment:
We can probably refactor this to have some method that check whether the field name matches some existing set of these known classes to ignore.